src/ol/collection.js

```diff
@@ -225,6 +249,20 @@ ol.Collection.prototype.updateLength_ = function() {
 
 
 /**
+ * @private
+ * @param {T} elem Element.
+ * @param {number=} opt_except Optional index to ignore.
+ */
+ol.Collection.prototype.assertUnique_ = function(elem, opt_except) {
+  for (var i = 0, ii = this.array_.length; i < ii; ++i) {
+    if (this.array_[i] === elem && i !== opt_except) {
+      throw new Error('Duplicate item added to a unique collection');
```

Comment:
Looks like `ol.asserts.assert()` isn't helpful when using the `ol` package since we don't provide `ol.VERSION`.  I'll leave this as something we can address separately.